service/javadsl/api/src/main/scala/com/lightbend/lagom/internal/javadsl/api/ServiceReader.scala

```diff
@@ -221,8 +221,18 @@ object ServiceReader {
   }
 
   private def constructServiceCallHolder[Request, Response](serviceCallResolver: ServiceCallResolver, method: Method): ServiceCallHolder = {
+
     val serializers = method.getGenericParameterTypes.toSeq.map { arg =>
-      serviceCallResolver.resolvePathParamSerializer(new UnresolvedTypePathParamSerializer[AnyRef], arg)
+      try {
+        serviceCallResolver.resolvePathParamSerializer(new UnresolvedTypePathParamSerializer[AnyRef], arg)
+      } catch {
+        case ex: IllegalArgumentException =>
+          throw new RuntimeException(s"Error encountered while resolving the ${method.getDeclaringClass + "." + method.getName}" +
```

Comment:
I think this is changing the behaviour of the code: what used to throw an `IllegalArgumentException` now throws a `RuntimeException`. The fact that the the build in Travis passed makes me think we're missing tests for failure cases.

Reply:
@ignasi35 Yes, it seems the test case was missing , I've added it.

Reply:
`IllegalArgumentException` is a `RuntimeException` and we were throwing it before. We should keep it as it was. 
We can still catch it to add a more meaning message, but we should keep it as `IllegalArgumentException`.

Reply:
@renatocaval Ok I changed it back to IllegalArgumentException as suggested.